src/uct/tcp/tcp_cm.c

```diff
@@ -156,6 +156,9 @@ ucs_status_t uct_tcp_cm_send_event(uct_tcp_ep_t *ep, uct_tcp_cm_conn_event_t eve
                             UCT_TCP_CM_CONN_ACK |
                             UCT_TCP_CM_CONN_WAIT_REQ)),
                 "ep=%p", ep);
+    ucs_assertv(!(ep->ctx_caps & UCS_BIT(UCT_TCP_EP_CTX_TYPE_TX)) ||
```

Comment:
why ||, not &&?

Reply:
if it does not have TX capability (i.e. only RX capability) and it is in CONNECTED state, it has to be able to send CM REQ event to a peer in order to set RX capability for the peer